daily_read/order_portal.py

```diff
@@ -100,43 +103,52 @@ def process_orders(self, priority, closed_before_in_days=30):
                     "recents": {},
                     "events": [],
                     "projects": {},
+                    "delete_report_for": {},
                 }
 
             order_updates_item = order_updates[proj_info.orderer]
-            order_updates_item["events"] += proj_info.events
+            if delete_report:
+                order_updates_item["delete_report_for"].setdefault(proj_info.status, []).append(proj_info)
+            else:
+                order_updates_item["events"] += proj_info.events
 
-            # Sort the statuses based on date and priority of event and extract the 5 first (done repeatedly for each new project added)
+                # Sort the statuses based on date and priority of event and extract the 5 first (done repeatedly for each new project added)
 
-            def sorting_key(item):
-                """Outputs date and the priority label for the status."""
-                date, (status, _) = item
-                return date, priority[status]
+                def sorting_key(item):
+                    """Outputs date and the priority label for the status."""
+                    date, (status, _) = item
+                    return date, priority[status]
 
-            orderer_recents = sorted(order_updates_item["events"], reverse=True, key=sorting_key)[:5]
+                orderer_recents = sorted(order_updates_item["events"], reverse=True, key=sorting_key)[:5]
 
-            order_updates_item["recents"] = orderer_recents
-            order_updates_item["projects"].setdefault(proj_info.status, []).append(proj_info)
-            order_updates_item["active_projects"] += 1
+                order_updates_item["recents"] = orderer_recents
+                order_updates_item["projects"].setdefault(proj_info.status, []).append(proj_info)
+                order_updates_item["active_projects"] += 1
 
         return order_updates
 
-    def upload_report_to_order_portal(self, report, project):
-        """Upload report to order portal"""
+    def upload_report_to_order_portal(self, report, project, status):
+        """Upload report to order portal
+        With the status 'published' the user can see the report immediately
+        With the status 'review', the user will not be able to view the report(used here as a proxy for deletion)
```

Comment:
So I guess it's not possible to delete a report then?

Reply:
Reports can't be deleted through the API currently, but only through the UI. We could ask for the functionality to be added, I guess

Reply:
Yes sounds like a good thing to ask for